### PR TITLE
Fill out Signature['Args'] (Part 2)

### PR DESCRIPTION
**Pull request by ijlee2**
## Description

Partially implements #27. In the `update-signatures` step, we can now partially fill out `Signature['Args']` for the end-developer.

Note, the codemod currently uses the empty object `{}` to indicate that the component doesn't receive any arguments. Glint, on the other hand, [recommends omitting the `'Args'` key](https://typed-ember.gitbook.io/glint/environments/ember/component-signatures) to record the absence.

> `Args` represents the arguments your component accepts. Typically this will be an object type mapping the names of your args to their expected type. If no `Args` key is specified, it will be a type error to pass any arguments to your component.

In a later task, we will want to refactor code to follow Glint's recommendations.

### CODE REVIEW

File: src/steps/update-signatures/builders.ts [hunk 1]
```diff
@@ -31,6 +31,26 @@ function builderConvertTsTypeToKeyword(tsType: string) {
   }
 }
 
+export function builderCreateArgsNode(signature: Signature) {
+  const members: unknown[] = [];
+
+  (signature.Args ?? []).forEach((argumentName) => {
```
Comment (ijlee2) [inline src/steps/update-signatures/builders.ts:37]: **TODO**

Currently, `signature.Args` can be `undefined` if the template file is not present. I don't think this is a valid case to consider.

Refactor code so that `signature.Args` is always of type `string[]`.

File: src/steps/update-signatures/update-signature.ts [hunk 2]
```diff
@@ -39,7 +40,14 @@ export function updateSignature(file: string, data: Data): string {
         return false;
       }
 
-      const ArgsNode = getBodyNode(path.node, 'Args');
+      const argsNode = getBodyNode(path.node, 'Args');
+
+      const isArgsKnown =
+        argsNode?.typeAnnotation.typeAnnotation.members.length > 0;
```
Comment (ijlee2) [inline src/steps/update-signatures/update-signature.ts:46]: **TODO**

This condition works if `Signature['Args']` is an object `{}` or the `'Args'` key is missing from `Signature`.

Check if there are edge cases for which the condition evaluates incorrectly or causes a runtime error.

File: tests/fixtures/ember-container-query-glint/output/src/components/widgets/widget-1.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import templateOnlyComponent from '@ember/component/template-only';
 
 interface WidgetsWidget1Signature {
+  Args: {};
```
Comment (ijlee2) [inline tests/fixtures/ember-container-query-glint/output/src/components/widgets/widget-1.ts:4]: **TODO**

Ideally, the output files in `ember-container-query-glint` should match the input files exactly. See if we can later remove `Args` from `Signature` when the component does not receive arguments.

File: tests/fixtures/ember-container-query-glint/output/src/components/widgets/widget-2.ts [hunk 4]
```diff
@@ -9,7 +9,6 @@ import {
 } from '../../utils/components/widgets/widget-2';
 
 interface WidgetsWidget2Signature {
-  // eslint-disable-next-line @typescript-eslint/ban-types
```
Comment (ijlee2) [inline tests/fixtures/ember-container-query-glint/output/src/components/widgets/widget-2.ts:12]: Removing the disable-comment may become an issue on the end-developer's side.

File: src/steps/update-signatures/update-signature.ts (hunk 2 above)
Comment (ijlee2) [reply src/steps/update-signatures/update-signature.ts:46]: Completed in #36.

Comment (ijlee2) [reply src/steps/update-signatures/update-signature.ts:46]: Resolved in #36.
